plugin/pkg/auth/authenticator/request/x509/x509.go

```diff
@@ -104,7 +105,13 @@ var DNSNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate) (
 
 // EmailAddressUserConversion builds user info from a certificate chain using the first EmailAddress on the certificate
 var EmailAddressUserConversion = UserConversionFunc(func(chain []*x509.Certificate) (user.Info, bool, error) {
```

Comment:
Authentication over TLS can use 1 of 3 fields within the user certificate to determine what is the username of the authenticating user. By default, it will use the CommonName field. If you want to use the 2 other fields (EmailAddress and DNSName), you will need to modify the code to replace `CommonNameUserConversion` with either `EmailAddressUserConversion` or `DNSNameUserConversion`.